### PR TITLE
Update for GNOME 44

### PR DESCRIPTION
**Pull request by DaPigGuy**
Updated gtk-rs crates to latest versions & bumped Flatpak runtime to 44

Seems updating the runtime to 44 had the side effect of removing the checkerboard pattern from the start of my recordings :slightly_smiling_face:

### CODE REVIEW

**Comment by DaPigGuy** [comment]
@SeaDve Could this get a review? The Flathub stable release causes undesired checkerboard frames at the start of recordings

**Comment by SeaDve** [comment]
Thank you for the PR, and also sorry for the really late update

File: src/area_selector/mod.rs [hunk 1]
```diff
@@ -160,6 +160,7 @@ impl AreaSelector {
 
             let monitor_geometry = RootExt::display(transient_for)
                 .monitor_at_surface(&transient_for.surface())
+                .expect("No monitor found")
```
Comment (SeaDve) [inline src/area_selector/mod.rs:163]: This could be the following instead, so it won't panic:
```suggestion
                .context("No monitor found")?
```

File: src/audio_device.rs [hunk 2]
```diff
@@ -34,7 +34,7 @@ pub async fn find_default_name(class: Class) -> Result<String> {
         .context("Failed to push future to main thread pool")?
         .await
     {
-        Ok(res) => Ok(res),
+        Ok(res) => res,
```
Comment (SeaDve) [inline src/audio_device.rs:37]: The entire function body could be simplify too to the following, so we can drop `THREAD_POOL` too:

```rust
pub async fn find_default_name(class: Class) -> Result<String> {
    match gio::spawn_blocking(move || find_default_name_gst(class))
        .await
        .map_err(|err| anyhow!("Failed to spawn blocking task: {:?}", err))?
    {
        Ok(res) => Ok(res),
        Err(err) => {
            tracing::warn!("Failed to find default name using gstreamer: {:?}", err);
            tracing::debug!("Manually using libpulse instead");

            let pa_context = pa::Context::connect().await?;
            pa_context.find_default_device_name(class).await
        }
    }
}
```

File: src/settings.rs [hunk 3]
```diff
@@ -191,16 +165,6 @@ fn present_message(heading: &str, body: &str, transient_for: Option<&impl IsA<gt
     dialog.present();
 }
 
-fn is_accessible(path: &Path) -> bool {
```
Comment (SeaDve) [inline src/settings.rs:194]: Why was this removed?

File: src/application.rs [hunk 4]
```diff
@@ -136,7 +136,7 @@ impl Application {
 
     async fn try_show_uri(&self, uri: &str) {
         if let Err(err) =
-            gtk::show_uri_full_future(self.main_window().as_ref(), uri, gdk::CURRENT_TIME).await
+            gtk::FileLauncher::new(Some(&gio::File::for_uri(uri))).launch_future(self.main_window().as_ref()).await
```
Comment (SeaDve) [inline src/application.rs:139]: This could be maybe separated to multiple lines

File: build-aux/io.github.seadve.Kooha.Devel.json [hunk 5]
```diff
@@ -1,11 +1,11 @@
 {
     "id": "io.github.seadve.Kooha.Devel",
     "runtime": "org.gnome.Platform",
-    "runtime-version": "43",
+    "runtime-version": "44",
     "sdk": "org.gnome.Sdk",
     "sdk-extensions": [
         "org.freedesktop.Sdk.Extension.rust-stable",
-        "org.freedesktop.Sdk.Extension.llvm14"
+        "org.freedesktop.Sdk.Extension.llvm15"
```
Comment (SeaDve) [inline build-aux/io.github.seadve.Kooha.Devel.json:8]: Changing this also requires the test targets `PATH` on `src/meson.build` files

**Review by SeaDve — CHANGES_REQUESTED** [review]
Other than that and a rustfmt, LGTM

File: build-aux/io.github.seadve.Kooha.Devel.json (hunk 5 above)
Comment (SeaDve) [reply build-aux/io.github.seadve.Kooha.Devel.json:8]: This is also likely why the CI is failing

File: Cargo.toml [hunk 6]
```diff
@@ -14,19 +14,19 @@ tracing = "0.1.36"
 tracing-subscriber = "0.3.15"
 once_cell = "1.9.0"
 gettext-rs = { version = "0.7.0", features = ["gettext-system"] }
-gtk = { package = "gtk4", version = "0.5.4" }
-gdk-wayland = { package = "gdk4-wayland", version = "0.5.4" }
-gdk-x11 = { package = "gdk4-x11", version = "0.5.4" }
-adw = { package = "libadwaita", version = "0.2.1", features = ["v1_2"] }
-gst_video = { package = "gstreamer-video", version = "0.19.4" }
-gst_pbutils = { package = "gstreamer-pbutils", version = "0.19.2", features = [
+gtk = { package = "gtk4", version = "0.6.6", features = ["v4_10"] }
```
Comment (SeaDve) [inline Cargo.toml:17]: You could also use `"gnome_44"` as feature instead, so it will be automatic with other libs too

```suggestion
gtk = { package = "gtk4", version = "0.6.6", features = ["gnome_44"] }
```

File: src/settings.rs (hunk 3 above)
Comment (DaPigGuy) [reply src/settings.rs:194]: The file chooser now gives access to the selected folder when sandboxed

Comment (SeaDve) [reply src/settings.rs:194]: The problem I had before is the portal returns path that starts with `/run` when the path is only accessible through the portal. Since we save the path to the settings, it is no longer accesible too once we relaunch the application, or restart the portal process.

I'm not sure if thus bug still exists, though.

Comment (DaPigGuy) [reply src/settings.rs:194]: It didn't seem to be the case when I had tested by both closing the app and restarting Fedora entirely

Comment (SeaDve) [reply src/settings.rs:194]: As far as I remember, It will only be reproducible when you have the app installed, instead of running from builder or vscode.

You also have to select folders that Kooha doesn't have permission to like Documents folder

Comment (DaPigGuy) [reply src/settings.rs:194]: Yes I had the flatpak exported and then installed

File: src/settings.rs [hunk 7]
```diff
@@ -33,57 +28,36 @@ impl Default for Settings {
...
-    /// Opens a `FileChooserDialog` to select a folder and updates
+    /// Opens a `FileDialog` to select a folder and updates
     /// the settings with the selected folder.
     pub fn select_saving_location(&self, transient_for: Option<&impl IsA<gtk::Window>>) {
-        let chooser = gtk::FileChooserDialog::builder()
+        let dialog = gtk::FileDialog::builder()
             .modal(true)
-            .action(gtk::FileChooserAction::SelectFolder)
             .title(&gettext("Select Recordings Folder"))
+            .accept_label(&gettext("_Select"))
```
Comment (SeaDve) [inline src/settings.rs:37]: This seems to be not needed by default

File: src/settings.rs [hunk 8]
```diff
@@ -33,57 +28,36 @@ impl Default for Settings {
...
-        }));
+        dialog.select_folder(
+            transient_for,
+            gio::Cancellable::NONE,
+            clone!(@weak inner, @weak parent => move |response| {
+                match response.ok().and_then(|file| file.path()) {
+                    Some(directory) => {
+                        inner.set("saving-location", &directory).unwrap();
+                    }
+                    None => {
```
Comment (SeaDve) [inline src/settings.rs:51]: I tested it, but it seems to also show the `No folder selected` error dialog even when cancelled. 

I think we can do the following:
```rust
let transient_for = transient_for.as_ref();
let inner = &self.0;
dialog.select_folder(
    Some(transient_for),
    gio::Cancellable::NONE,
    clone!(@weak inner, @weak transient_for => move |response| {
        match response {
            Ok(folder) => {
                inner.set("saving-location", folder.path().unwrap()).unwrap();
            }
            Err(err) => {
                if err.matches(gtk::DialogError::Dismissed) || err.matches(gtk::DialogError::Cancelled) {
                    return;
                }

                present_message(
                    &gettext("Failed to select folder"),
                    &err.to_string(),
                    Some(&transient_for),
                );
            }
        }
    }),
);
```

File: src/settings.rs (hunk 3 above)
Comment (SeaDve) [reply src/settings.rs:194]: It now works too on my side, it probably has been fixed on the portal side

File: src/settings.rs [hunk 9]
```diff
@@ -33,57 +28,39 @@ impl Default for Settings {
 impl Settings {
     pub const NONE_PROFILE_ID: &'static str = "none";
 
-    /// Opens a `FileChooserDialog` to select a folder and updates
+    /// Opens a `FileDialog` to select a folder and updates
     /// the settings with the selected folder.
     pub fn select_saving_location(&self, transient_for: Option<&impl IsA<gtk::Window>>) {
```
Comment (SeaDve) [inline src/settings.rs:33]: Since transient_for is unwrapped anyway at the function body, it could instead just take in &impl Is<gtk::Window> without the options

**Review by SeaDve — APPROVED** [review]

**Comment by DaPigGuy** [comment]
Kooha is now having issues recording videos on both the main branch and this branch. Looking into the cause right now

**Comment by SeaDve** [comment]
We could also have this merge first, then maybe do it in a separate PR

**Comment by DaPigGuy** [comment]
That works too :slightly_smiling_face:

**Comment by SeaDve** [comment]
Merged. Thanks again for the PR!

**Comment by DaPigGuy** [comment]
Recording issues seem to have fixed themselves at some point, on Fedora 38 at least.
